crates/sp-domains-fraud-proof/src/host_functions.rs

```diff
@@ -118,6 +116,38 @@ where
         .map(|ext| ext.encode())
     }
 
+    fn derive_domain_set_code_extrinsic(
+        &self,
+        consensus_block_hash: H256,
+        domain_id: DomainId,
+    ) -> Option<Option<Vec<u8>>> {
```

Comment:
`Option<Option<T>>` is valid sometimes, but maybe create an enum that will be easier to read with 3 variants?